modules/client/shared/src/main/scala/com/azavea/stac4s/api/client/SttpStacClientF.scala

```diff
@@ -151,20 +149,48 @@ case class SttpStacClientF[F[_]: MonadThrow, S: Lens[*, Option[PaginationToken]]
 }
 
 object SttpStacClientF {
+  // TODO: should be a newtype
+  type PaginationToken = NonEmptyString
```

Comment:
That is probably not a part of this PR and could be a more broad discussion. Good for now!